lib/aws-ssm-env/parameter_setter.rb

```diff
@@ -0,0 +1,45 @@
+module AwsSsmEnv
+  # @author Dominic O'Connor
+  # @since 0.1.2
+  class ParameterSetter
+    def initialize(**args)
+      @overwrite = overwrite?(args)
+      @scope = scope?(args)
+    end
+
+    def save(name, value)
+      if @scope[name] && !@overwrite
+        return
+      end
+      @scope[name] = value
+    end
+
+    def scope
+      @scope
+    end
+
+    private
+
+    def overwrite?(overwrite: nil, **)
+      if overwrite.nil?
+        false
+      else
+        overwrite.to_s.downcase == 'true'
+      end
+    end
+
+    def scope?(**args)
+      scope_type = args[:scope]
+      case scope_type
+      when nil
+        ENV
+      else
+        unless scope_type.is_a?(OpenStruct)
```

Comment:
@dominic-oconnor 
How about judging whether there is a `[]=` method instead of limiting it to `OpenStruct` ?

```ruby
unless scope_type.respond_to?(:[]=)
  raise ArgumentError, '...'
end
```